Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Review/DTOs/Validators/UpdateReviewValidator.cs

```diff
@@ -0,0 +1,20 @@
+﻿using BlogApp.Application.Contracts.Persistence;
+using FluentValidation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlogApp.Application.Features.Review.DTOs.Validators
+{
+    public class UpdateReviewValidator: AbstractValidator<ReviewDto>
+    {
+        public UpdateReviewValidator(IUnitOfWork unitOfWork)
+        {
+            Include(new IReviewValidator());
+            RuleFor(p => p.Id)
+            .MustAsync(async (id, token) => await unitOfWork.BlogRepository.Exists(id)).WithMessage($"Blog not found");
```

Comment:
Here the Id field on the review entity represents the id of the review not of blogs...it makes sesnse to check if there is a review by the id already sent so you should change the Blog exist test to review exist test.

--> Maybe you could also check if the blogId is chenged or not while updating a review(this can be done in your updateReviewCommandHandler method! this is because if you let the user change a blogId in his review he might create a new review for a new blog(which we don't want happening!))